fontir/src/ir.rs

```diff
@@ -1028,7 +1028,7 @@ pub struct Glyph {
     pub emit_to_binary: bool,
     pub codepoints: HashSet<u32>, // single unicodes that each point to this glyph. Typically 0 or 1.
     default_location: NormalizedLocation,
-    sources: HashMap<NormalizedLocation, GlyphInstance>,
+    instances: HashMap<NormalizedLocation, GlyphInstance>,
```

Comment:
why the attribute name change? These are called sources in DesignSpace jargon, instances usually mean the obtain from interpolating.. the sources

Reply:
From the IRs point of view they are instances. The source terminology is only applicable to UFO+ds and thus struck me as not belonging.

Reply:
to me at least it's confusing to call the sources "instances", and I'm sure a lot of font developers would feel the same.

Reply:
FWIW I thought I'd find the instances in something with instance in the name, "sources" actually delayed me. It's relatively normal to name collections after what's in them in my experience. A bunch of GlyphInstances is instances, "sources" requires domain knowledge to grok.

I also thought, perhaps mistakenly, that was leaking a specific ... source (UFO+ds)... into our abstraction over many source formats. 

If none of that is convincing I can change it back :) Actually probably what I should do is back it out here and submit as a different PR (if you are convinced) since it's relatively unrelated to the rest of this.

Reply:
I see what you mean about the names of collections but I am not convinced. If anything I'd propose to rename GlyphInstance struct. In all font formats and font editors, a master.. ehm sorry a "source" is the input that the designer draws, a whole value that is associated with some location in the variation space, and is used for... interpolating instances, which are the output or the result of the interpolation. This is unversally the case.

Reply:
> instances usually mean the thing you obtain from interpolating.. the sources

Thinking "aloud" I really like the name GlyphInstance, I think it captures what it is very accurately. If I pursue this line of reasoning I think i would conclude it's the wrong name and I don't like that conclusion. That would lead me to argue IR simply doesn't use "instance" in this way.

Reply:
just try and ask in the font engs chat and see what they mean by "instance"

Reply:
I have undone the naming change. I would be quite strongly opposed to renaming GlyphInstance. I don't think the IR aims to model what an editor interacts with. It's meant to support a SWE like a Google Fonts member and to that group "something instance" generally has meaning, the correct meaing, whereas "source" is a rather broad term that quite likely means the wrong thing.

I read Fontra's VariableGlyph as aligning with our "Glyph" and StaticGlyph as aligning with GlyphInstance. .glyphs just has "glyphs". Needing to lookup a layer to find a location appears to me to be a product of how a user-facing editor works: you create a layer with a location then draw on it. I don't think that concept really applies to IR, we wouldn't gain anything that I can currently identify by modelling layers.

I propose we decide to rename nothing. GlyphInstance is a good name and my world won't end if I have to find the glyph instances in "sources"

Reply:
> I read Fontra's VariableGlyph as aligning with our "Glyph" and StaticGlyph as aligning with GlyphInstance

right, as I said Fontra's VariableGlyph has "sources", not "instances".

> meant to support a SWE like a Google Fonts member and to that group "something instance" generally has meaning, the correct meaing

whatever :)

> I propose we decide to rename nothing.

agreed!

Reply:
> "source" is a rather broad term that quite likely means the wrong thing

it highlights the _function_. It's "example or specific occurrence" (instance) of a glyph that is designed (not generated) and is positioned along variation axes in order to obtain other such instances through interpolation. If I go to fonttools or glyphsLib codebase and find/replace all occurrences of the word "master" (a banned word that we replaced with "source") with "instance", it surely will raise a few eyebrows.

Reply:
It's indeed about the function. A "source" is a designed source of (glyph/kerning/metrics/whatever) data at a location in a designspace, aka a "master value". An "instance" is an interpolation at any location in a designspace. Since they can both be represented with the same type I came up with `StaticGlyph` as the object type, to contrast with `VariableGlyph`, and then it is less weird to use the same object type for both functions. I agree that `GlyphInstance` is a bit weird for a "master value", but `GlyphSource` would definitely be weird for an instance.

Reply:
It may take some getting used to, but I think ultimately "source" is a pretty great synonym for "master". (I think this meaning was introduced by Erik van Blokland in the .designspace file format.)

Reply:
I would prefer the entire concept of "master" get dropped as we push towards sparse definitions. This glyph has definition at {some set of locations}, this other one has them at {some other set}.

I don't see a need for IR to distinguish whether an example/occurance/instance is designed source or not, it's just the definition at a location. IR is a bridge between source formats and binary, not an editors representation.

For context, the IR currently uses XInstance consistently (GlyphInstance, KerningInstance, GlobalMetricsInstance, NamedInstance) to mean a thing defined at a specific location in designspace. That makes "sources" as a collection of them seem weird, something doesn't line up right. If we wanted to "fix" XInstance naming - I don't think we should, but let's pursue it just to see where it takes us - then perhaps what would make sense would be to strip Instance suffixes and instead have a naming convention for things that are variable. For example, instead of Glyph/GlyphInstance you'd have VariableGlyph/Glyph. 

I still think I land at don't actually change anything :)